src/unstract/sdk/ocr.py

```diff
@@ -11,28 +12,43 @@
 
 
 class OCR(metaclass=ABCMeta):
-    def __init__(self, tool: BaseTool):
+    def __init__(
+        self,
+        tool: BaseTool,
+        adapter_instance_id: str,
+    ):
         self.tool = tool
         self.ocr_adapters = adapters
+        self.adapter_instance_id = adapter_instance_id
+        self.ocr_instance: OCRAdapter = self._get_ocr()
 
-    def get_ocr(self, adapter_instance_id: str) -> Optional[OCRAdapter]:
+    def _get_ocr(self) -> Optional[OCRAdapter]:
         try:
             ocr_config = ToolAdapter.get_adapter_config(
-                self.tool, adapter_instance_id
+                self.tool, self.adapter_instance_id
             )
             ocr_adapter_id = ocr_config.get(Common.ADAPTER_ID)
             if ocr_adapter_id in self.ocr_adapters:
-                ocr_adapter = self.ocr_adapters[ocr_adapter_id][
-                    Common.METADATA
-                ][Common.ADAPTER]
+                ocr_adapter = self.ocr_adapters[ocr_adapter_id][Common.METADATA][
+                    Common.ADAPTER
+                ]
                 ocr_metadata = ocr_config.get(Common.ADAPTER_METADATA)
-                ocr_adapter_class = ocr_adapter(ocr_metadata)
+                self.ocr_instance = ocr_adapter(ocr_metadata)
 
-                return ocr_adapter_class
+                return self.ocr_instance
 
         except Exception as e:
             self.tool.stream_log(
-                log=f"Unable to get OCR adapter {adapter_instance_id}: {e}",
+                log=f"Unable to get OCR adapter {self.adapter_instance_id}: {e}",
                 level=LogLevel.ERROR,
             )
             return None
```

Comment:
@gaya3-zipstack We should catch specific errors always. We could iteratively improve later.